tests/integration/llm/client.py

```diff
@@ -1078,16 +1078,19 @@ def test_handler_adapters(model, model_spec):
     res = requests.delete(
         f"http://127.0.0.1:8080/models/test/adapters/{del_adapter}")
     logging.info(f"del adapter {res}")
-    res = send_json(reqs[0]).content.decode("utf-8")
-    logging.info(f"call deleted adapter {res}")
-    if "error" not in res:
+
+    try:
+        # Following send_json should fail with a 424 for calling a deleted adapter
+        res = send_json(reqs[0]).content.decode("utf-8")
+        logging.info(f"call deleted adapter {res}")
+    except ValueError:
```

Comment:
I think we should change the logic in send_json, or call requests.post directly here so that we can directly validate the response status and code. It's possible that the request fails for another reason/status here, but still passes.

Reply:
Updated to narrow the error conditions